File: sense_motion_mqtt.py
import time
import paho.mqtt.client as mqtt

# The callback for when the client receives a CONNACK response from the server.
def on_connect(client, userdata, flags, rc):
    print("Connected with result code " + str(rc))


client = mqtt.Client()
client.on_connect = on_connect
client.connect("iot.eclipse.org", 1883, 60)

# Come up with a unique id to use, e.g., the last portion of the IP address
id = ...
topic = "RHSummit2015_temp_rpi_HC-SR501/" + id

# Motion callback now sends a message to the MQTT broker, customize if you wish
def MOTION(PIR_PIN):
    now = time.time()
    json = "{{'event':'motion-detection', 'time': {:f}}}".format(now)
    print json
    client.publish(topic, json)

# Copy the sense_motion.py code for the motion detection to here...

File: sense_motion_mqtt.js
var Gpio = require('onoff').Gpio;
// Configure GPIO pin 21 for input and rising edge detection
var pir = new Gpio(21,'in','rising');

// Setup the mqtt client
var mqtt = require('mqtt');
var client = mqtt.connect('mqtt://iot.eclipse.org:1883")
// Come up with a unique id to use, e.g., the last portion of the IP address
var id = ...
var topic = "RHSummit2015_temp_rpi_HC-SR501/" + id

client.on('connect', function () {
    console.log("Connected");
});

// Add the edge detection callback to catch the motion detection events
pir.watch(function(err, value) {
  if (value === 1) {
    // The pin went high
    console.log("Motion Detected: %d", value);
    // Publish the motion-detected event to the broker
    var date = new Date();
    client.publish(topic, "{'event':'motion-detected', 'time': "+date.getTime()+"}", 1);
  }
});

function exit() {
  console.log("Exiting");
  pir.unexport();
  process.exit();
}

process.on('SIGINT', exit);

console.log("Monitoring...");
